### PR TITLE
examples: slipdev integration example

### LINKED ISSUE #85: Add slipdev example
### Description

Itś necessary implement the slipdev example to have some knowledge how it can use the net protocols under UART

### PR DESCRIPTION
**Pull request by eduazocar**
### Contribution description
This PR it's to add the slipdev example, to use the UART as net device.

### Testing procedure

when you are using pyterm console in your board, use:
ifconfig
```sh
> ifconfig
2022-01-13 14:30:46,017 # ifconfig
2022-01-13 14:30:46,020 # Iface  4 
2022-01-13 14:30:46,024 #           Long HWaddr: AE:4F:B5:37:F8:1D:6C:EF 
2022-01-13 14:30:46,027 #           MTU:65535  HL:64  
2022-01-13 14:30:46,029 #           Link type: wired
2022-01-13 14:30:46,035 #           inet6 addr: fe80::ac4f:b537:f81d:6cef  scope: link  VAL
2022-01-13 14:30:46,038 #           inet6 group: ff02::1
2022-01-13 14:30:46,041 #           inet6 group: ff02::1:ff1d:6cef
2022-01-13 14:30:46,042 #           
2022-01-13 14:30:46,045 #           Statistics for Layer 2
2022-01-13 14:30:46,048 #             RX packets 9  bytes 468
2022-01-13 14:30:46,052 #             TX packets 49 (Multicast: 0)  bytes 2580
2022-01-13 14:30:46,056 #             TX succeeded 0 errors 0
2022-01-13 14:30:46,058 #           Statistics for IPv6
2022-01-13 14:30:46,061 #             RX packets 9  bytes 468
2022-01-13 14:30:46,066 #             TX packets 49 (Multicast: 4)  bytes 2580
2022-01-13 14:30:46,069 #             TX succeeded 49 errors 0
2022-01-13 14:30:46,070 # 
```
then use ping to the link local address of the another board that is connected in the UART. in this example it's used fe80::e405:df5c:7fc4:2314 to the secondary board.
```sh
ping fe80::e405:df5c:7fc4:2314
```
expected outpu after pingt:
```sh
2022-01-13 14:30:59,292 # 12 bytes from fe80::e405:df5c:7fc4:2314%4: icmp_seq=0 ttl=64 time=10.384 ms
2022-01-13 14:31:00,291 # 12 bytes from fe80::e405:df5c:7fc4:2314%4: icmp_seq=1 ttl=64 time=9.962 ms
2022-01-13 14:31:01,291 # 12 bytes from fe80::e405:df5c:7fc4:2314%4: icmp_seq=2 ttl=64 time=9.970 ms
2022-01-13 14:31:01,292 # 
2022-01-13 14:31:01,296 # --- fe80::e405:df5c:7fc4:2314 PING statistics ---
2022-01-13 14:31:01,301 # 3 packets transmitted, 3 packets received, 0% packet loss

```

### Issues/PRs references
fixes #85

### CODE REVIEW

File: examples/esp32-wroom-32/slipdev/Makefile [hunk 1]
```diff
@@ -0,0 +1,47 @@
+APPLICATION = slipdev_test
```
Comment (luisan00) [inline examples/esp32-wroom-32/slipdev/Makefile:1]:
```suggestion
APPLICATION = slipdev_example
```

File: examples/esp32-wroom-32/slipdev/main.c [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ */
+
+/**
+ * @ingroup     examples/esp32-wroom
```
Comment (luisan00) [inline examples/esp32-wroom-32/slipdev/main.c:13]:
```suggestion
 * @ingroup     examples_esp32-wroom
```
Why not the current form?

File: examples/esp32-wroom-32/slipdev/main.c [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+#include "msg.h"
+
+#define MAIN_QUEUE_SIZE     (8)
+static msg_t _main_msg_queue[MAIN_QUEUE_SIZE];
+
+static const shell_command_t shell_commands [] = {
+    {NULL, NULL, NULL}
+};
+
+
```
Comment (luisan00) [inline examples/esp32-wroom-32/slipdev/main.c:34]: too many empty things

**Review by luisan00 — CHANGES_REQUESTED** [review]
good job man,  only some minor things

File: examples/esp32-wroom-32/slipdev/slipdev_params.h [hunk 4]
```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2015 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
```
Comment (luisan00) [inline examples/esp32-wroom-32/slipdev/slipdev_params.h:10]: i think this should be changed , i don't know if it fit in our documentation build system

File: examples/esp32-wroom-32/slipdev/README.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
+# Esp32-wroom examples
```
[the remark this answers is not in the document]
Comment (luisan00) [reply examples/esp32-wroom-32/slipdev/README.md:1]: yeah, looks a mistake

**Comment by xkevin190** [comment]
looks good, but you not added the  EOF space of the end every file bro.

**Comment by xkevin190** [comment]
Screen Shot 2022-01-14 at 11 03 09 AM

missing EOF space

**Comment by xkevin190** [comment]
Screen Shot 2022-01-14 at 11 03 19 AM

missing EOF space

File: examples/esp32-wroom-32/slipdev/main.c [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+#define MAIN_QUEUE_SIZE     (8)
+msg_t _main_msg_queue[MAIN_QUEUE_SIZE];
+
+int main(void)
+{
+    msg_init_queue(_main_msg_queue, MAIN_QUEUE_SIZE);
+    char line_buf[SHELL_DEFAULT_BUFSIZE];
+    shell_run(NULL, line_buf, SHELL_DEFAULT_BUFSIZE);
+}
```
Comment (xkevin190) [inline examples/esp32-wroom-32/slipdev/main.c:35]: missing EOF space

**Review by xkevin190 — CHANGES_REQUESTED** [review]

**Review by xkevin190 — APPROVED** [review]

**Review by luisan00 — APPROVED** [review]

**Comment by luisan00** [comment]
Let's go ahead @eduazocar

**Comment by luisan00** [comment]
please squash before merge bro

**Pull request merged**